src/Config/ObjectModel/McpDmlToolsOptions.cs

```diff
@@ -0,0 +1,154 @@
+        }
+
+        if (ExecuteRecord is not null)
+        {
+            this.ExecuteRecord = (bool)ExecuteRecord;
+            UserProvidedExecuteRecord = true;
+        }
+        else
+        {
+            this.ExecuteRecord = false;
+        }
+    }
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write describe-entities
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the describe-entities property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(DescribeEntities))]
+    public bool UserProvidedDescribeEntities { get; init; } = false;
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write create-record
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the create-record property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(CreateRecord))]
+    public bool UserProvidedCreateRecord { get; init; } = false;
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write read-record
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the read-record property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(ReadRecord))]
+    public bool UserProvidedReadRecord { get; init; } = false;
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write update-record
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the update-record property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(UpdateRecord))]
+    public bool UserProvidedUpdateRecord { get; init; } = false;
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write delete-record
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the delete-record property/value, which signals DAB to use the default,
+    /// the DAB CLI should not write the default value to a serialized config.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.Always)]
+    [MemberNotNullWhen(true, nameof(DeleteRecord))]
+    public bool UserProvidedDeleteRecord { get; init; } = false;
```

Comment:
if neither DeleteRecord nor UserProvidedDeleteRecord are nullable, do we need the qualification - `MemberNotNullWhen`?  its going to be always NOT NULL.